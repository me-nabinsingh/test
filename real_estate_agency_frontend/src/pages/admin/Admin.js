import React, { Component} from 'react';
import {Router, Route, Switch } from "react-router-dom";

import Header from '../../components/admin/header/Header';
import Sidebar from '../../components/admin/sidebar/Sidebar';
import Dashboard from './dashboard/Dashboard';

import BannerList from './banners/BannerList';
import NewBanner from './banners/NewBanner';
import EditBanner from './banners/EditBanner';

import PlanetList from './planets/PlanetList';
import NewPlanet from './planets/NewPlanet';
import EditPlanet from './planets/EditPlanet';

import ImageList from './images/ImageList';
import NewImage from './images/NewImage';

import history from "../../util/history";

class  Admin extends Component {

    render() {
        return ( 
            <Router history={history}>
                <div>
                    <Header />
                    <div className="container-fluid">
                        <div className="row">
                            <Sidebar />
                            <main role="main" className="col-md-9 ml-sm-auto col-lg-10 px-4 py-2 mt-5">
                                   <Switch>
                                        <Route exact path="/admin/" component={Dashboard} />
                                        
                                        <Route exact path="/admin/banners" component={BannerList} />
                                        <Route exact path="/admin/banners/new" component={NewBanner} />
                                        <Route exact path="/admin/banners/:id" component={EditBanner} />

                                        <Route exact path="/admin/planets" component={PlanetList} />
                                        <Route exact path="/admin/planets/new" component={NewPlanet} />
                                        <Route exact path="/admin/planets/:id" component={EditPlanet} />

                                        <Route exact path="/admin/images" component={ImageList} />
                                        <Route exact path="/admin/images/new" component={NewImage} />

                                    </Switch> 
                            </main>
                        </div>
                    </div>
                </div>
            </Router>
         );
    }
}
 
export default Admin;