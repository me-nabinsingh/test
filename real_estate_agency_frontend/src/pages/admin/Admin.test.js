import React from 'react';
import Admin from './Admin';
import renderer from 'react-test-renderer';
import {Router } from "react-router-dom";
import history from "../../util/history";

describe('<Admin />', () => {
    it('renders without crashing', () => {
        const component = renderer.create(
            <Router history={history}>
                <Admin />
            </Router>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});