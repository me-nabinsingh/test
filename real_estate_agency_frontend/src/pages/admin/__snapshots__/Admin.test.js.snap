// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> renders without crashing 1`] = `
<div>
  <nav
    className="navbar navbar-dark fixed-top bg-dark flex-md-nowrap p-0 shadow"
  >
    <div
      className="mx-5 my-2"
    >
      <a
        className="py-2"
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          height={50}
          src="logo.svg"
          width={50}
        />
      </a>
    </div>
    <ul
      className="navbar-nav px-3"
    >
      <li
        className="nav-item text-nowrap"
      >
        <a
          className="nav-link"
          href="/auth/login"
          onClick={[Function]}
        >
          Sign out
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <nav
        className="col-md-2 d-none d-md-block bg-light sidebar"
      >
        <div
          className="sidebar-sticky  mt-5"
        >
          <ul
            className="nav flex-column py-2"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/admin/"
                onClick={[Function]}
              >
                Dashboard
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/admin/banners"
                onClick={[Function]}
              >
                Banners
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/admin/planets"
                onClick={[Function]}
              >
                Planets
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/admin/images"
                onClick={[Function]}
              >
                Images
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <main
        className="col-md-9 ml-sm-auto col-lg-10 px-4 py-2 mt-5"
        role="main"
      />
    </div>
  </div>
</div>
`;
